vcd/resource_vcd_vapp_firewall_rules_test.go

```diff
@@ -0,0 +1,505 @@
+					resource.TestCheckResourceAttr(resourceName, "rule.2.protocol", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.destination_port", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.destination_vm_ip_type", "assigned"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.destination_vm_nic_id", "0"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.source_port", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.source_vm_nic_id", "0"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.source_vm_ip_type", "NAT"),
+					resource.TestCheckResourceAttr(resourceName, "rule.2.enable_logging", "false"),
+
+					testAccCheckVcdVappFirewallRulesExists(resourceName, params["Name4"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.policy", "drop"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.protocol", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.destination_port", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.destination_ip", "external"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.source_port", "any"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.source_ip", "internal"),
+					resource.TestCheckResourceAttr(resourceName, "rule.3.enable_logging", "true"),
+
+					testAccCheckVcdVappFirewallRulesExists(resourceName+"2", params["Name1"].(string)),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.policy", "drop"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.protocol", "udp"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.destination_port", "221"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.destination_ip", "any"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.source_port", "any"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.source_ip", "10.10.0.10/24"),
+					resource.TestCheckResourceAttr(resourceName+"2", "rule.0.enable_logging", "false"),
+				),
+			},
+			resource.TestStep{ // Step 1 - resource import
+				ResourceName:            "vcd_vapp_firewall_rules.imported",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateIdFunc:       importStateVappFirewallRuleObject(testConfig, vappName, vappNetworkName),
+				ImportStateVerifyIgnore: []string{"network_id", "org", "vdc"},
+			},
+			resource.TestStep{ // Step 2 - resource import by ID
+				ResourceName:            "vcd_vapp_firewall_rules.imported2",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateIdFunc:       importStateVappFirewallRuleById(testConfig, resourceName),
+				ImportStateVerifyIgnore: []string{"network_id", "org", "vdc"},
```

Comment:
What are the reasons to ignore `network_id`. Would it stay unset after import?

Reply:
(map[string]string) (len=1) {
         (string) (len=10) "network_id": (string) (len=55) "urn:vcloud:network:b51f8e4e-db7f-400f-8d43-e87dc4638375"
        }
        
        
        (map[string]string) (len=1) {
         (string) (len=10) "network_id": (string) (len=36) "b51f8e4e-db7f-400f-8d43-e87dc4638375"
        }
        
Due I have to work with two types of Id's

Reply:
Please elaborate why do you have two types of IDs? Where is one and the other coming from?

Reply:
change to use normalizeId